adhocracy4/modules/views.py

```diff
@@ -0,0 +1,45 @@
+from django.core.urlresolvers import reverse
+from django.shortcuts import get_object_or_404
+from django.views import generic
+
+from adhocracy4.rules import mixins as rules_mixins
+
+from . import models
+
+
+class ModuleDetailView(rules_mixins.PermissionRequiredMixin,
+                       generic.DetailView):
+    model = models.Module
+    permission_required = 'a4projects.view_project'
+    slug_url_kwarg = 'module_slug'
+
+    def dispatch(self, request, *args, **kwargs):
+        module = self.get_object()
+        if module.is_active:
+            return ModuleRedirectView.as_view()(request, *args, **kwargs)
```

Comment:
Note: This only works if there is only one active module at any time. This could be changed by checking "is this the module that is currently displayed on the project page" instead.